debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+                    "The field containing the timestamp, or empty if the entire value is a timestamp")
+            .define(TARGET_TYPE_CONFIG, ConfigDef.Type.STRING, ConfigDef.Importance.LOW,
+                    "The desired timestamp representation: string, unix, Date, Time, or Timestamp")
+            .define(TIMESTAMP_FORMAT_CONFIG, ConfigDef.Type.STRING, TIMESTAMP_FORMAT_DEFAULT, ConfigDef.Importance.LOW,
+                    "A SimpleDateFormat-compatible format for the timestamp. Used to generate the output when type=string "
+                            + "or used to parse the input if the input is a string.")
+            .define(DATE_FORMAT_CONFIG, ConfigDef.Type.STRING, DATE_FORMAT_DEFAULT, ConfigDef.Importance.LOW,
+                    "A SimpleDateFormat-compatible format for the date. Used to generate the output when type=string "
+                            + "or used to parse the input if the input is a string.")
+            .define(FIELD_TYPE_CONFIG, ConfigDef.Type.STRING, FIELD_TYPE_DEFAULT, ConfigDef.Importance.LOW, "is a comma seperated string, providing an option to add multiple converters.\n" +
+                    "<Type of Timestamp Object> -> <To Target type>\n" +
+                    "can add multiple type converters as below," +
+                    " <Type of Timestamp Object> -> <To Target type>,<Type of Timestamp Object> -> <To Target type>")
+            .define(STRUCT_FIELD_NAME_CONFIG, ConfigDef.Type.STRING, STRUCT_FIELD_NAME_DEFAULT, ConfigDef.Importance.LOW, "Field name that is only to be modified in Nested Struct");
+
+    private static final String PURPOSE = "converting timestamp formats";
+    private static final String TYPE_INT64 = "INT64";
+    private static final String TYPE_INT32 = "INT32";
+    private static final String TYPE_INT16 = "INT16";
+    private static final String TYPE_INT8 = "INT8";
+    private static final String TYPE_FLOAT32 = "FLOAT32";
+    private static final String TYPE_FLOAT64 = "FLOAT64";
+    private static final String TYPE_BOOLEAN = "BOOLEAN";
+    private static final String STRING_TYPE = "STRING";
+    private static final String TYPE_BYTES = "BYTES";
+    private static final String TYPE_STRUCT = "STRUCT";
+
+    private static final String TYPE_STRING = "string";
+    private static final String TYPE_UNIX = "unix";
+    private static final String TYPE_DATE = "Date";
+    private static final String TYPE_TIME = "Time";
+    private static final String TYPE_TIMESTAMP = "Timestamp";
+    private static final String TYPE_EPCOH_DATE = "EpochDate";
+    private static final Set<String> VALID_TYPES = new HashSet<>(Arrays.asList(TYPE_STRING, TYPE_UNIX, TYPE_DATE, TYPE_TIME, TYPE_TIMESTAMP, TYPE_EPCOH_DATE));
+
+    private static final TimeZone UTC = TimeZone.getTimeZone("UTC");
+
+    private static final Map<String, TimestampTranslator> TRANSLATORS = new HashMap<>();
+    private static final Map<String, OptionalTypeFieldSchema> OPTIONAL_TRANSLATORS = new HashMap<>();
```

Comment:
Why not just `Map<String, Schema>` and remove the `OptionalTypeFieldSchema` interface altogether? Or, if you really need to have an object that can create a non-constant `Schema` instance, use `Map<String, Supplier<Schema>>` instead.